stdlib/runtime/numbers.gr

```diff
@@ -405,6 +437,35 @@ export let coerceNumberToWasmI32 = (x: Number) => {
         }
         WasmI32.wrapI64(int64)
       },
+      t when WasmI32.eq(t, Tags._GRAIN_BIGINT_BOXED_NUM_TAG) => {
+        BI.toInt32(x)
+      },
+      _ => {
+        // rationals are never integral, and we refuse to coerce floats to ints
+        throw Exception.NumberNotIntlike
```

Comment:
Do we refuse to avoid losing precision?

Reply:
Yes, precisely. Floating point numbers are, by construction, "inexact", so it would be improper (in my opinion, and, IIRC, in the Racket folks' opinions) to coerce them to an "exact" type like an integer.